front/app/components/admin/ActionForm/FlowVisualization/Blocks/SSOBlock.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+
+import {
+  Box,
+  colors,
+  stylingConsts,
+  Icon,
+  Text,
+} from '@citizenlab/cl2-component-library';
+import styled from 'styled-components';
+
+import useVerificationMethodVerifiedActions from 'api/verification_methods/useVerificationMethodVerifiedActions';
+
+import useLocalize from 'hooks/useLocalize';
+
+import { useIntl } from 'utils/cl-intl';
+
+import messages from './messages';
+import SSOConfigModal from './SSOConfigModal';
+import {
+  getVerificationFrequencyExplanation,
+  getReturnedFieldsPreview,
+} from './utils';
+
+const StyledBox = styled(Box)`
+  &:hover {
+    background-color: ${colors.teal100};
+  }
+`;
+
+interface Props {
+  verificationExpiry: number | null;
+  onChangeVerificationExpiry: (value: number | null) => void;
+}
+
+const SSOBlock = ({
+  verificationExpiry,
+  onChangeVerificationExpiry,
+}: Props) => {
+  const { formatMessage } = useIntl();
+  const localize = useLocalize();
+  const { data: verificationMethod } = useVerificationMethodVerifiedActions();
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const verificationMethodMetadata =
+    verificationMethod?.data.attributes.action_metadata;
+
+  if (!verificationMethodMetadata) return null;
+
+  const authenticateMessage = formatMessage(
+    messages.authenticateWithVerificationProvider,
+    {
+      verificationMethod: verificationMethodMetadata.name,
+    }
+  );
+
+  const verifiedFieldsMessage = formatMessage(messages.verifiedFields);
+
+  const verificationFrequencyExplanation = getVerificationFrequencyExplanation(
+    verificationExpiry,
+    formatMessage
+  );
+
+  const returnedFieldsPreview = getReturnedFieldsPreview(
+    verificationMethodMetadata,
+    localize
+  );
+
+  return (
+    <>
+      <StyledBox
+        borderRadius={stylingConsts.borderRadius}
+        border={`1px solid ${colors.blue700}`}
+        bgColor={colors.teal50}
+        p="16px"
+        w="240px"
+        style={{
+          cursor: 'pointer',
+        }}
+        onClick={() => setModalOpen(true)}
```

Comment:
It's probably better to convert this into a button if it has an onClick handler. It will make it accessible by keyboard. Alternatively, we should at least add a tab index to it to make it focusable and make sure the enter key works as a click.